cloudinit/net/netplan.py

```diff
@@ -48,7 +48,7 @@ def _get_params_dict_by_match(config, match):
     )
 
 
-def _extract_addresses(config: dict, entry: dict, ifname, features=None):
+def _extract_addresses(config: dict, entry: dict, ifname, features: Callable):
```

Comment:
Rather than taking a callable here, whose typing will be fairly opaque, can we instead perform the call on line 476 so that this takes a `List[str]` instead?

Reply:
This would make us incur the cost of calling `netplan apply` in many cases that we don't need it. I would rather only call it when needed, given how costly this is.